sdk/core/core-http/lib/url.ts

```diff
@@ -176,6 +178,42 @@ export class URLBuilder {
     return this._scheme;
   }
 
+  /**
+   * Get the username that has been set in this URL.
+   */
+  public getUsername(): string | undefined {
+    return this._username;
+  }
+
+  /**
+   * Set the username that has been set in this URL.
+   */
+  public setUsername(username: string | undefined) {
+    if (!username) {
+      this._username = undefined;
+    } else {
+      this.set(username, "AUTH_OR_HOST");
+    }
+  }
+
+  /**
+   * Get the password that has been set in this URL.
+   */
+  public getPassword(): string | undefined {
+    return this._password;
+  }
+
+  /**
+   * Set the password that has been set in this URL.
+   */
+  public setPassword(password: string | undefined) {
+    if (!password) {
+      this._password = undefined;
+    } else {
+      this.set(password, "PASSWORD");
```

Comment:
feels weird to have a default password

Reply:
"PASSWORD" is a state in the state machine (`set` is not a good name but I didn't created it...)

Reply:
hmm, actually it's a token type

Reply:
"PASSWORD" is used for both.  Here it's the current state